component/kafka/kafka.go

```diff
@@ -0,0 +1,73 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/correlation"
+	"github.com/google/uuid"
+)
+
+// FailStrategy type definition.
+type FailStrategy int
+
+const (
+	// ExitStrategy does not commit failed message offsets and exits the application.
+	ExitStrategy FailStrategy = iota
+	// SkipStrategy commits the offset of messages that failed processing, and continues processing.
+	SkipStrategy
+)
+
+// BatchProcessorFunc definition of a batch async processor.
+type BatchProcessorFunc func(context.Context, []MessageWrapper) error
```

Comment:
In the SQS package we expose the Batch interface which exposes []Messages. We should follow the same pattern for uniformity.